### ISSUE TITLE
Issue with Null Exception in AbstractConfigFactory

### ISSUE THREAD

**Issue by mikeedwards83** [mikeedwards83/Glass.Mapper#172]
This was moved from Issue 137

We were seeing a similar stack trace. Restarting the web app was the only resolution. This stack trace was from 4.0.1.5

```
Exception: System.NullReferenceException
Message: Object reference not set to an instance of an object.
Source: Glass.Mapper
   at Glass.Mapper.IoC.AbstractConfigFactory`1.<GetItems>b__0(Func`1 f)
   at System.Linq.Enumerable.WhereSelectListIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.<ReverseIterator>d__a0`1.MoveNext()
   at Glass.Mapper.Pipelines.AbstractPipelineRunner`2..ctor(IEnumerable`1 tasks)
   at Glass.Mapper.AbstractService..ctor(Context glassContext)
   at Glass.Mapper.Sc.ItemExtensions.GlassCast[T](Item item, Boolean isLazy, Boolean inferType)
   at ASP._Page_Areas_XXXXX_Views_Layouts_DefaultLayout_cshtml.Execute() in c:\inetpub\XXXXX\Website\Areas\XXXXX\Views\Layouts\DefaultLayout.cshtml:line 78
   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy()
   at System.Web.Mvc.WebViewPage.ExecutePageHierarchy()
   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy(WebPageContext pageContext, TextWriter writer, WebPageRenderingBase startPage)
   at System.Web.Mvc.Html.PartialExtensions.Partial(HtmlHelper htmlHelper, String partialViewName, Object model, ViewDataDictionary viewData)
   at Sitecore.Mvc.Presentation.ViewRenderer.Render(TextWriter writer)
```

I've upgraded the solution to 4.0.3.44-beta and will be testing and (ideally pushing to production).

cc: @nicolas-stephens

**Comment by cardinal252** [comment]
I have added a couple of 'educated guess fixes' but we cannot replicate this issue easily. These should be out when the next dev build goes up.

**Comment by mikeedwards83** [comment]
Potential fix for 172 as been rolled into development. Nightly build 4.0.3.46

**Comment by devilishgoat** [comment]
I have tested the fix that you have put together on our live solution and it is working.  Thanks for your help

**Comment by cardinal252** [comment]
Great news! :D - could you maybe try to replicate tomorrow again to confirm so we can get it into the next release.

Cheers

**Comment by mikeedwards83** [comment]
This issue has been fixed in release 4.0.3.50
